mm2src/lp_ordermatch.rs

```diff
@@ -3285,6 +3360,129 @@ pub async fn set_price(ctx: MmArc, req: Json) -> Result<Response<Vec<u8>>, Strin
+                    None => return ERR!("Rel coin {} has been removed from config", rel),
+                };
+
+                let original_conf_settings = order.conf_settings.unwrap();
+
+                let updated_conf_settings = OrderConfirmationsSettings {
+                    base_confs: req.base_confs.unwrap_or(original_conf_settings.base_confs),
+                    base_nota: req.base_nota.unwrap_or(original_conf_settings.base_nota),
+                    rel_confs: req.rel_confs.unwrap_or(original_conf_settings.rel_confs),
+                    rel_nota: req.rel_nota.unwrap_or(original_conf_settings.rel_nota),
+                };
+
+                (
+                    base_coin,
+                    rel_coin,
+                    order.price.clone(),
+                    order.max_base_vol.clone(),
+                    updated_conf_settings,
+                    order.matches.clone(),
+                )
+            },
+            None => return ERR!("There is no order with UUID {}", req.uuid),
+        };
+
+    drop(my_maker_orders);
+
+    let mut update_msg = new_protocol::MakerOrderUpdated::new(req.uuid);
+    update_msg = update_msg.with_new_conf_settings(updated_conf_settings);
+
+    // Validate and Add new_price to update_msg if new_price is found in the request
+    let new_price = match req.new_price {
+        Some(new_price) => {
+            try_s!(validate_price(new_price.clone()));
+            update_msg = update_msg.with_new_price(new_price.clone().into());
+            new_price
+        },
+        None => original_price,
+    };
+
+    let min_base_amount = base_coin.min_trading_vol();
+    let min_rel_amount = rel_coin.min_trading_vol();
+
+    // Add min_volume to update_msg if min_volume is found in the request
+    if let Some(min_volume) = req.min_volume.clone() {
+        // Validate and Ca lculate Minimum Volume
+        let actual_min_vol = try_s!(validate_and_get_min_vol(
+            min_base_amount.clone(),
+            min_rel_amount.clone(),
+            Some(min_volume),
+            new_price.clone()
+        ));
+        update_msg = update_msg.with_new_min_volume(actual_min_vol.into());
+    }
+
+    // Calculate order volume and add to update_msg if new_volume is found in the request
+    let new_volume = if req.max.unwrap_or(false) {
+        let max_volume = try_s!(get_max_volume(&ctx, &base_coin, &rel_coin).await);
+        update_msg = update_msg.with_new_max_volume(max_volume.clone().into());
+        max_volume
+    } else if Option::is_some(&req.new_volume) {
+        try_s!(
+            check_balance_for_maker_swap(
+                &ctx,
+                &base_coin,
+                &rel_coin,
+                req.new_volume.clone().unwrap(),
+                None,
+                None,
+                FeeApproxStage::OrderIssue
+            )
+            .await
+        );
+        update_msg = update_msg.with_new_max_volume(req.new_volume.clone().unwrap().into());
```

Comment:
I've noticed 1 more thing :slightly_smiling_face: If the order is already matched partially at the time of the update its available volume for trading (`MakerOrder::available_amount`) will be less than what is set in the request. I'm even a bit unsure of how we should handle it from the UX perspective:
1. Should we consider the `new_volume` to be fully available for trading. So `max_base_vol = available_amount + new_volume`.
1. Or we simply update the `max_base_vol` so `new_available = new_volume - matched_amount`

Reply:
In my opinion, I think the second case `new_available = new_volume - matched_amount` is the right one since the user when updating the order he is thinking of the total volume for the order including the matched volume and it will be similar to the `set_price` call in this case.
A solution for both might be adding a field to the request specifying if the user wants to add the `new_volume` to the old one or changing the total volume but it might add unneeded complexity to the request.

Reply:
Yeah, it looks fine actually as we have `available_volume` in the response. The only fix we require is we should use `available_volume` in the `MakerOrderUpdated` message instead of the `new_volume` so other nodes are getting correct volumes.
 @shamardy Could you please make this change?

Reply:
>     2. Or we simply update the `max_base_vol` so `new_available = new_volume - matched_amount`?

I would also go for this solution and in order to keep it simple, i would offer only this option. If the user wants the other behavior, he can cancel the old order completely and make a new one.
The only potential problem i see with `matched_amount` is that multiple partial matches can occur while order is up, so mm2 would need to keep track of all matches and increase `matched_amount` every time a match occurred.

Reply:
> The only potential problem i see with matched_amount is that multiple partial matches can occur while order is up, so mm2 would need to keep track of all matches and increase matched_amount every time a match occurred.

@cipig This part is already handled :slightly_smiling_face:

Reply:
`MakerOrder::available_amount` is not a field - it's always calculated "on the fly" using the matches that the order already has.